tests/unit/via/views/test_route_by_content.py

```diff
@@ -17,52 +17,23 @@
 
 
 class TestRouteByContent:
-    @pytest.mark.parametrize(
-        "requested_path,expected_location,content_type",
-        [
-            # If the requested pdf URL has no query string then it should just
-            # redirect to the requested URL, with no query string (but with the
-            (
-                "/https://example.com/foo",
-                "http://localhost/pdf/https://example.com/foo",
-                "application/pdf",
-            ),
-            # If the requested pdf URL has a query string then the query string
-            # should be preserved in the URL that it redirects to.
-            (
-                "/https://example.com/foo?bar=baz",
-                "http://localhost/pdf/https://example.com/foo?bar=baz",
-                "application/pdf",
-            ),
-            # If the requested html URL has a query string then the query string
-            # should be preserved in the URL that it redirects to.
-            (
-                "/https://example.com/foo?bar=baz",
-                "http://via.hypothes.is/https://example.com/foo?bar=baz",
-                "text/html",
-            ),
-            # If the requested html URL has a client query params then the query params
-            # should be preserved in the URL that it redirects to.
-            (
-                "/https://example.com/foo?via.open_sidebar=1",
-                "http://via.hypothes.is/https://example.com/foo?via.open_sidebar=1",
-                "text/html",
```

Comment:
All of these tests amount to the same thing, and are replicated lower down. We always pass all params onto the final request. It only changes when we call the original URL to check the resource content type.